src/Gizmos/gizmo.ts

```diff
@@ -187,16 +189,27 @@ export class Gizmo implements IDisposable {
 
         if  ((<Camera>this._attachedNode)._isCamera) {
             var camera = this._attachedNode as Camera;
+            var worldMatrix;
+            var worldMatrixUC;
             if (camera.parent) {
-                var parentInv = new Matrix();
-                var localMat = new Matrix();
-                camera.parent.getWorldMatrix().invertToRef(parentInv);
-                this._attachedNode.getWorldMatrix().multiplyToRef(parentInv, localMat);
-                localMat.decompose(this._tempVector2, this._tempQuaternion, this._tempVector);
+                var parentInv = this._tempMatrix2;
+                camera.parent._worldMatrix.invertToRef(parentInv);
+                this._attachedNode.getWorldMatrix().multiplyToRef(parentInv, this._tempMatrix1);

```

Comment:
Don't you want to also use `this._attachedNode._worldMatrix` here?

Reply:
Good catch!